service_container.rst

```diff
@@ -852,7 +852,7 @@ This is mostly useful when you want to fetch services lazily::
 
 As a best practice, you should only create *private* services, which will happen
 automatically. And also, you should *not* use the ``$container->get()`` method to
-fetch public services.
+fetch private services.
```

Comment:
How about:

```
As a best practice, you should only create *private* services.
This allows for safe container optimizations, e.g. removing
unused services. You should not use ``$container->get()``
to fetch public services, as it will make it harder to make those
services private later. Instead consider :ref:`injecting services <_services-constructor-injection>`
or using :doc:`Service Subscribers or Locators </service_container/service_subscribers_locators>`.

I prefer this over the old text, because it not only tells you not to do it, but also why and what alternatives you have. What do you think?

Reply:
Totally agree, really better especially because it tells you why you should do that.